### PR TITLE
Rename version fields and more

### PR DESCRIPTION
**Pull request by Hartmnt**
Hi, sorry this MR took so long (health+life issues).

I am posting this unfinished, untested and possibly non-compiling merge request to have a dedicated communication channel.

In the first commit, I renamed the version fields a little. I also noticed that Message uses CamelCase while the proto uses underscores, which I also did not want to break.

In the later commit, I tried to check the version against a const ``UnknownVersion`` but that is when I realized the following:
I understand ``Message`` is an abstraction of the ``.proto`` fields for use in the custom clients/servers. But why don't we add a ``Version`` struct to ``Message`` (removing the versionV1 and versionV2 fields) and keep the conversion from raw numbers (and also v1->v2 v2->v1) to libmumble?
I do not think servers and clients using libmumble need to explicitly deal with the different version formats. Or Is there something I am missing here? I would try to implement that, if it is desired @davidebeatrici

### CODE REVIEW

**Comment by davidebeatrici** [comment]
Hi, no worries about the delay. Hope you're good now.

@Krzmbrzl Actually made a similar suggestion regarding the version abstraction: https://github.com/mumble-voip/libmumble/commit/021fe1af6179d808924fd91b25f9f2163a0124b9

The idea is to use the following `struct`:

https://github.com/mumble-voip/libmumble/blob/9a6917036eba8a642ab973910e8997e6a63be5bf/include/mumble/Types.hpp#L59-L105

And eventually move the "blob" part into a private header.

**Comment by Hartmnt** [comment]
@davidebeatrici This MR now represents what I had in mind. This seems to be consistent with @Krzmbrzl annotations. Please let me know (in a sentence or two), how to implement a private header such that ``Version::blob32`` and ``Version::blob64`` can not be accessed by the user of the API (or rather how you would like it implemented).

P.S I had quite some debugging trouble because message structs in ``Messages.hpp`` containing other message structs (such as ``UserState``) did not use fully qualified type names. In this case I confused the ``Version`` member with ``mumble::Version``, but ``mumble::tcp::Message::Version`` was actually used. Maybe we should consider using fully qualifying types when dealing with messages within messages.

**Comment by Krzmbrzl** [comment]
> Please let me know (in a sentence or two), how to implement a private header such that Version::blob32 and Version::blob64 can not be accessed by the user of the API (or rather how you would like it implemented).

Simply add the header in the source directory instead of in the include directory. That means that the public API will never see this header (but also that it can't use any types defined in it).

> Maybe we should consider using fully qualifying types when dealing with messages within messages.

Yeah, that sounds like a good idea.

**Comment by Krzmbrzl** [comment]
In that case you'd have to define two headers: One public header, without the blob stuff and on private header implementing the functions for converting to and from blobs

**Comment by davidebeatrici** [comment]
@Hartmnt Sorry, I missed your message from 5 days ago because clicking on the notification gets you to the updated diff instead of the conversation.

The functions that @Krzmbrzl suggested could also be put in a dedicated namespace or perhaps even in `Pack` (name subject to change).

**Comment by Hartmnt** [comment]
@Krzmbrzl @davidebeatrici ``blob32`` is used in [the example server](https://github.com/mumble-voip/libmumble/blob/master/src/example/ExampleServer/Node.cpp#L341) for the legacy ping field. I am not sure how to refactor this, to make the blob functions private...

**Comment by davidebeatrici** [comment]
We could make an exception for `blob32()` and move it into the `legacy` namespace.

**Comment by Hartmnt** [comment]
I almost forgot that this was still open. I do not think there is an elegant way to move the ``blob`` functions out of the struct in the public header. At least I can not think of one. So as far as I am concerned this MR could be merged as is.

If you have any concrete change you want me to implement, feel free to let me know.

**Comment by davidebeatrici** [comment]
I think it's fine to merge it as it is.

File: include/mumble/Types.hpp [hunk 1]
```diff
@@ -87,6 +87,10 @@ struct Version {
 			   | (std::min< uint32_t >(patch, std::numeric_limits< uint8_t >::max()) << offsetPatch32);
 	}
 
+	bool isValid() const {
+		return (major | minor | patch | extra) != 0;
+	}
```
Comment (davidebeatrici) [inline include/mumble/Types.hpp:92]:
```suggestion
	constexpr bool isValid() const {
		return (major | minor | patch | extra) != 0;
	}
```

**Review by davidebeatrici — APPROVED** [review]

**Comment by Krzmbrzl** [comment]
> I think it's fine to merge it as it is.

Feel free to do so - I'm fine with that as well :+1:

**Comment by davidebeatrici** [comment]
Thank you very much!

**Pull request merged**